tests/cli/bootsys/test_platform.py

```diff
@@ -266,116 +290,124 @@ def fake_run_remote_command(command, *args, **kwargs):
         self.set_up_mock_run_remote_command(stop_timeout=True)
-        mock_get_containers = mock.patch.object(self.cst, '_get_running_containers').start()
+        mock_get_containers = mock.patch.object(self.csp, '_get_running_containers').start()
         # On first call, containers are running; on second call, there are none
         mock_get_containers.side_effect = [['a', 'b', 'c'], []]
 
         with self.assert_exits_successfully(f'All containers stopped on {self.host}.') as logs_cm:
-            self.cst.run()
+            self.csp.run()
 
         self.assertEqual(
             [mock.call('systemctl is-active containerd', err_on_non_zero=False),
              mock.call(CONTAINER_STOP_SCRIPT, err_on_non_zero=False)],
-            self.cst._run_remote_command.mock_calls
+            self.csp._run_remote_command.mock_calls
         )
-        warnings = [record for record in logs_cm.records if record.levelno == logging.WARNING]
-        self.assertEqual(1, len(warnings))
-        self.assertEqual(f'One or more "crictl stop" commands timed out on {self.host}',
-                         warnings[0].message)
         self.assertEqual([mock.call()] * 2, mock_get_containers.mock_calls)
 
-    def test_run_timeout_running_containers_exist(self):
-        """Test run method when the stop command times out, and containers remain running."""
-        self.set_up_mock_run_remote_command(stop_timeout=True)
-        mock_get_containers = mock.patch.object(self.cst, '_get_running_containers').start()
-        # On first call, 3 containers are running; on second call, one remains
-        mock_get_containers.side_effect = [['a', 'b', 'c'], ['c']]
-
-        err_msg = (f'Failed to stop 1 container(s) on {self.host}. Execute "crictl ps -q" '
-                   f'on the host to view running containers.')
+    def test_run_retry_stops_remaining_containers(self):
+        """Test run method when some containers remain running initially and are stopped in subsequent retries."""
+        # Create a mock for _get_running_containers method
+        mock_get_containers = mock.patch.object(self.csp, '_get_running_containers').start()
+
+        # Define the side effects for _get_running_containers:
+        # - On the first call, 4 containers ('a', 'b', 'c', 'd') are running
+        # - On the second call, 2 containers ('c', 'd') remain running
+        # - On the third call, 2 containers ('c', 'd') remain running
+        # - On the fourth call, no containers are running
+        mock_get_containers.side_effect = [['a', 'b', 'c', 'd'], ['c', 'd'], ['c', 'd'], []]
+
+        # Mock the _run_remote_command method
+        mock_run_command = mock.patch.object(self.csp, '_run_remote_command').start()
+
+        # Set up the side effects for _run_remote_command:
+        mock_run_command.side_effect = [
+            (0, 'active\n', ''),
+            (1, 'a\nb', ''),
+            (0, 'c\nd', ''),
```

Comment:
This call will actually be the call to `systemctl is-active containerd` during the second iteration of the `while` loop in the `run` method.

```suggestion
            (0, 'active\n', ''),
```

Reply:
Addressed